firmware/esp32/gdoor/src/mqtt_helper.cpp

```diff
@@ -73,23 +74,100 @@ namespace MQTT_HELPER { //Namespace as we can only use it once
     MQTTClient mqttClient; // MQTT Library object
 
     const char* rx_topic_name; // Which callback topic
+    const char* tx_topic_name; // For HA discovery message
     const char* user; // Username
     const char* password; // Password
 
     MQTT_PRINTER printer(&mqttClient); // Printer, so that code can use the Arduino print functions
 
     String received_mqtt_payload; // Global variable which stores received MQTT payload
     String empty(""); //Empty string, useful as global and fixed allocated value.
+    String availability_topic; //Topic where availabiliy is shown
 
     bool new_string_available = false; // Global variable to indicate that a new MQTT String was received
 
     bool newly_connected = true; // Global variable to indicate a newly established WIFI connection
+    bool new_connection_established = false; //Global variable to indicate we successfully connected new
+    bool ha_online = false; // Indicates if Home assistant messaged a new online state, so that we can resend our state
+
+    /**
+     * Function which sends home assistant discovery message,
+     * e.g.
+     * {
+     *  "name": "Bus Data",
+     *  "force_update": True,
+     *  "icon": "mdi:door",
+     *  "state_topic": "gdoor/rx",
+     *  "command_topic": "gdoor/tx",
+     *  "json_attributes_topic": "gdoor/rx",
+     *  "value_template": "{{ value_json.action }}",
+     *  "uniq_id": "gdoor_data_mac",
+     *  "device": {
+     *      "sw_version": "3.0",
```

Comment:
it would be nice if we can update the version number automatically with every build/release in https://github.com/gdoor-org/gdoor/issues/12.
Further, it would be nice if the version is also shown in the wifimanager portal.

Could we e.g. add a version file which contains the version string and read this file here instead of hardcoding it?

Reply:
Depends on how and where we create the release. Do we use the Git hash? Or manually?

Reply:
I'm planning to follow [these docs](https://github.com/softprops/action-gh-release?tab=readme-ov-file#-limit-releases-to-pushes-to-tags), meaning whenever a tag with the format `v*.*.*` is pushed, a release is build. During the build step we could use this tag name and use it as the version number. Appending the Git hash would also be possible, e.g. `3.1.0-3b9bac2`. Alternatively we could define the version number statically in the code and update it manually when we create the version tag.

Reply:
I would prefer something like `3.1.0-3b9bac2`:
1. As a user I do not like it but:
2. In case there is a release error, all releases will be unique by principle,
as a developer I like it very much ;-) (Even in Debian there are problems sometimes *g*)

I took a look, it seems the "tag name" (aka release number in our case) is in `github.ref_name`.

According to https://community.platformio.org/t/pass-global-define-at-run-time/1439/10
we can set DEFINES via the command line (via env. variables).

So I would say:
1. We set 
`#define GDOOR_VERSION "dev"`
in `defines.h`

3. We use `GDOOR_VERSION` wherever needed

4. We override `GDOOR_VERSION` value with `github.ref_name`+short hash in the platformIO release process.

This will set "normal" builds to "dev" but releases would contain the tag name.

If needed, I can do 1. and 2., just tell me if I should do it  :)

Reply:
Sounds good, please go ahead with 1 and 2! Thanks :-)